euclidlib/photo/_phz.py

```diff
@@ -2,43 +2,155 @@
 
 from collections.abc import Mapping
 from os import PathLike
-from typing import Any
+from typing import TYPE_CHECKING
 
 import fitsio  # type: ignore [import-not-found]
 import numpy as np
-from numpy.typing import NDArray
+
+if TYPE_CHECKING:
+    from typing import Any
+    from numpy.typing import ArrayLike, NDArray
+
+
+def _hist2dist(x: NDArray[Any], y: NDArray[Any]) -> NDArray[Any]:
+    """
+    Convert histogram to distribution.
+    """
+    *dims, n = y.shape
+    z = np.zeros((*dims, n + 1), dtype=y.dtype.base)
+    np.cumsum(y, axis=-1, out=z[..., 1:])
+    result: NDArray[Any] = np.gradient(z, x, axis=-1, edge_order=1)
+    return result
+
+
+def write_nz(
+    path: str | PathLike[str],
+    z: ArrayLike,
+    nz: ArrayLike,
+    *,
+    weight_method: str = "NO_WEIGHT",
+    bin_type: str = "TOM_BIN",
+    hist: bool = False,
+) -> None:
+    """
+    Write n(z) data in Euclid SGS format.  Supports both distributions
+    (when *hist* is false, the default) and histograms (when *hist* is
+    true).
+    """
+
+    z = np.asanyarray(z)
+    nz = np.asanyarray(nz)
+
+    if z.ndim != 1:
+        raise ValueError("z array must be 1D")
+    if nz.ndim == 0:
+        raise ValueError("nz array must be at least 1D")
+    if not hist and z.shape[-1] == nz.shape[-1]:
+        pass
+    elif hist and z.shape[-1] == nz.shape[-1] + 1:
+        pass
+    else:
+        raise ValueError("shape mismatch between redshifts and values")
+
+    # PHZ uses a fixed binning scheme with z bins in [0, 6] and dz=0.002
+    zbinedges = np.linspace(0.0, 6.0, 3001)
+
+    # turn nz into a 2D array with NBIN rows
+    nz = nz.reshape(-1, nz.shape[-1])
+    nbin = nz.shape[0]
+
+    # create the output data in the correct format
+    out = np.empty(
```

Comment:
Same with the name of the variable `out`, it could be more specific.

Reply:
That's the common name for the return array of numpy ufuncs.